pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go

```diff
@@ -498,10 +498,10 @@ func deployNeededInstallation(
 
 	if controllerInstallationName != "" {
 		// The installation already exists, however, we do not have the latest version of the ControllerInstallation object.
-		// Hence, we are running the `CreateOrUpdate` function as it first GETs the current objects and then runs the mutate()
-		// function before sending the UPDATE. This way we ensure that we have applied our mutations to the latest version.
+		// Hence, we are running the `GetAndCreateOrMergePatch` function as it first GETs the current objects and then runs the
+		// mutate() func before sending the PATCH. This way we ensure that we have applied our mutations to the latest version.
 		controllerInstallation.Name = controllerInstallationName
-		_, err := controllerutil.CreateOrUpdate(ctx, c, controllerInstallation, mutate)
```

Comment:
> This comment sounds like something else outside of this controller is also modifying the controllerinstallation

I don't think so. It is just created with `metadata.generateName` (see a few lines below), so we can't know the name upfront. But we need it for getting the object before patching it, when executing reconciliation again.

> In case of a stale cache do we run into the danger of overwriting changes here if we do not rely on optimistic locking?

Nope. The mutate func only touches the spec and specific labels (doesn't overwrite all labels like in the comment below), so only those parts will be overwritten. But they are also exclusively owned by this controller, so overwriting is fine and actually the right thing to do (see reasoning above).